packages/application-shell-connectors/src/components/project-extension-image-regex/project-extension-image-regex.tsx

```diff
@@ -54,24 +73,36 @@ const ProjectExtensionProviderForImageRegex = (props: ProviderProps) => {
 ProjectExtensionProviderForImageRegex.displayName =
   'ProjectExtensionProviderForImageRegex';
 
-const GetProjectExtensionImageRegex = (props: ConsumerProps) => (
-  <Context.Consumer>
-    {(imageRegexContext) => props.render(imageRegexContext)}
-  </Context.Consumer>
-);
+const GetProjectExtensionImageRegex = (props: ConsumerProps) => {
+  //TODO: add a descriptive possible deprecation warning for moving from HOC to Hooks in the future
+  useWarning(
+    false,
+    `@commercetools-frontend/application-shell-connectors: ...`
+  );
+
+  return (
+    <Context.Consumer>
+      {(imageRegexContext) => props.render(imageRegexContext)}
+    </Context.Consumer>
+  );
+};
 GetProjectExtensionImageRegex.displayName = 'GetProjectExtensionImageRegex';
 
 function withProjectExtensionImageRegex<Props extends {}>(
-  propKey = 'imageRegexData'
+  propKey: 'imageRegexData'
 ) {
   return (Component: ComponentType<Props>) => {
-    const WrappedComponent = (props: Props) => (
-      <GetProjectExtensionImageRegex
-        render={(imageRegexData) => (
-          <Component {...props} {...{ [propKey]: imageRegexData }} />
-        )}
-      />
-    );
+    const WrappedComponent = (props: Props) => {
+      const imageregexContext = useProjectExtensionImageRegex();
```

Comment:
Maybe we can also use a deprecation message with the HOC.
Perhaps something like this:
```
@commercetools-frontend/application-shell-connectors: It is not recommended to use the 'withProjectExtensionImageRegex' high order component anymore. Please use the 'useProjectExtensionImageRegex' hook instead.
```

Not 100% sure though since maybe we have customers who still use React class components and they wouldn't be able to use the hook.

Let's see what the rest of the team think about it.

Reply:
Yes, good point. I am not sure too but it would be nice to see what the team thinks too but I will also update it.